Tests/DatadogTests/Datadog/Mocks/DatadogCoreMock.swift

```diff
@@ -9,6 +9,11 @@ import Foundation
 
 internal final class DatadogCoreMock: Flushable {
     private var v1Features: [String: Any] = [:]
+    private var v1Context: DatadogV1Context?
```

Comment:
Can this be `internal` so we can mutate the context during a test case? it could then be reset to nil or `.mockAny()` on `flush`, WDYT?

Reply:
Good idea 👌, it helps a lot in some repetitive tests like `LoggerTests` where we need to dance with the `core` each time. 

Changed ✅, however, I left the `init(v1Context:)` too, as in some other tests the inline initialisation comes handy.